buildengine/testdata/type_registry_main.go

```diff
@@ -12,40 +13,35 @@ import (
 	"ftl/other"
 )
 
-func main() {
-	verbConstructor := server.NewUserVerbServer("other",
-		server.HandleCall(other.Echo),
-	)
-	ctx := context.Background()
-
-	tr := reflection.NewTypeRegistry(
+func init() {
+	reflection.Register(
 		reflection.WithSumType[another.SecondTypeEnum](
 			*new(another.One),
 			*new(another.Two),
 		),
-		reflection.WithSumType[another.TypeEnum](
-			*new(another.A),
-			*new(another.B),
-		),
 		reflection.WithSumType[other.SecondTypeEnum](
 			*new(other.A),
 			*new(other.B),
 		),
 		reflection.WithSumType[other.TypeEnum](
-			*new(other.Bool),
-			*new(other.Bytes),
-			*new(other.Float),
-			*new(other.Int),
-			*new(other.Time),
+			*new(other.MyBool),
+			*new(other.MyBytes),
+			*new(other.MyFloat),
+			*new(other.MyInt),
+			*new(other.MyTime),
```

Comment:
This all happened automatically. I'm not actually sure how the test that uses this was passing before because only List, Map, and Struct don't have the "My" prefix in the `other` module

Reply:
That test is haunted.
But more likely I broke it here sorry... https://github.com/TBD54566975/ftl/pull/1558